### PR TITLE
Add clustering config with tests

### PR DESCRIPTION
**Pull request by kirillsakh**
Add:

- config.yaml
- config.py
- tests
- ci.yml

### CODE REVIEW

File: src/config.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+
+class ConfigContainer(containers.DeclarativeContainer):
+    """Container for managing configuration settings."""
+
+    config = providers.Configuration()
+
+    get_algorithm = providers.Selector(
+        selector=config.clustering.type,
+        kmeans=providers.Factory(KMeans.create, config.clustering.kmeans),
```
Comment (tynaneu) [inline src/config.py]: you should write explicitly what parameters are expected in class initialization

Comment (tynaneu) [inline src/config.py]: KMeans.create should not be there - just KMeans

File: tests/test_config.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+from src.config import ConfigContainer, KMeans
+from src.file_handlers.json_handler import JsonHandler
+from src.file_handlers.numpy_handler import NumpyHandler
+
+from .data import CONFIG_DATA
+
+
+class TestConfigContainer:
+    def test_config_init(self, config_container: ConfigContainer):
```
Comment (tynaneu) [inline tests/test_config.py:14]: did you run the tests? I think they are not working.... It is a good practice to have test in CI/CD pipeline

Comment (kirillsakh) [reply tests/test_config.py:14]: I tested them locally and they worked for me. I will add a CI workflow to run tests in the next commit.

**Pull request merged**